From a3320429ffc85bd91b5abf010cfe6c0b05fbba51 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 23 Apr 2018 18:52:13 +0800
Subject: [PATCH 3683/5242] MLK-18031 video: mxsfb: enable global alpha when
 grayscale is 0 for 32bpp format

commit  f2c6e664a086e5789a8ec503fd8c6ad48361804a from
https://source.codeaurora.org/external/imx/linux-imx.git

For overlay framebuffer, when its grayscale is '0' and the format
bpp is 32, enable the global alpha blending by default which can
make the overlay fb can display the framebuffer content as long as
it is unblanked.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 4f40427..6f2d18f 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1663,6 +1663,7 @@ static void overlayfb_setup(struct mxsfb_layer *ofb)
 			break;
 		case 32: /* ARGB8888 */
 			format = 0x0;
+			global_alpha_en = 1;
 			break;
 		default:
 			return;
-- 
1.7.9.5

